pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -1558,6 +1571,60 @@ private void computeResultsForLiteral(Literal literal, List<String> columnNames,
     }
   }
 
+  private BrokerResponseNative processShowTablesQuery(long compilationStartTimeNs,
+      RequestContext requestContext) {
+    BrokerResponseNative brokerResponse = new BrokerResponseNative();
+    String[] columnNames = {"Tables"};
+    DataSchema.ColumnDataType[] columnTypes = {DataSchema.ColumnDataType.STRING};
+
+
+    DataSchema dataSchema = new DataSchema(columnNames, columnTypes);
+    List<Object[]> rows = new ArrayList<>();
+    _tableCache.getTableConfigs().stream().map(TableConfig::getTableName).sorted().forEach(a -> {
+      a = a.replace("_OFFLINE", "").replace("_REALTIME", "");
+      rows.add(new String[] {a});
+    });
+    ResultTable resultTable = new ResultTable(dataSchema, rows);
+    brokerResponse.setResultTable(resultTable);
+
+    long totalTimeMs = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - compilationStartTimeNs);
+    brokerResponse.setTimeUsedMs(totalTimeMs);
+    requestContext.setQueryProcessingTime(totalTimeMs);
+    augmentStatistics(requestContext, brokerResponse);
+    return brokerResponse;
+  }
+
+  private BrokerResponseNative processDescribeTableQuery(String tableName, long compilationStartTimeNs,
```

Comment:
We need to handle `database` and access control here

Reply:
I've added check for database tables access. Any other access control required?